### ISSUE TITLE
manager.py "view" locks up after extended duration running

### ISSUE THREAD

**Issue by Jaga-Telesin** [swar/Swar-Chia-Plot-Manager#1036]
Seen this on two machines now, one W10 and the other W7 SP2.  Don't have exact times after which it won't respond anymore.  It stops updating the powershell display, and won't respond to a CTRL-C anymore.  Have to close the powershell process entirely, sometimes requiring a restart on the manager.py process to get it going again.

Typically it's happening to me after periods >2 hours, but I was not tracking it.  Currently if I use "view" I let it display stats once, then I immediately CTRL-C the script to stop it.

**Comment by bathrobehero** [comment]
Same, I'm contemplating going back to regular batch scripts. It keeps freezing (have to kill python), or just shows abnormal values (no PID, etc,) and not launching more plots and it sometimes freezes while saying something about psutil.

And since I've started using this otherwise awesome tool, I get tons of plots that crash or simply disappear. Not sure if ocnnected.

**Comment by elforesto** [comment]
The most helpful thing to do is to create dump files and attach them to the issue. In Task Manager, right-click the appropriate python process and choose Create dump file. This should include useful debugging information.

**Comment by mah3re** [comment]
> Seen this on two machines now, one W10 and the other W7 SP2. Don't have exact times after which it won't respond anymore. It stops updating the powershell display, and won't respond to a CTRL-C anymore. Have to close the powershell process entirely, sometimes requiring a restart on the manager.py process to get it going again.
> 
> Typically it's happening to me after periods >2 hours, but I was not tracking it. Currently if I use "view" I let it display stats once, then I immediately CTRL-C the script to stop it.

Having the same exact issue as well from my very first run of swar manager, so I'm assuming that closing the cli will cancel all the plots? Also, if I were to just leave it alone do you know if it will still continue working?

**Comment by Jaga-Telesin** [comment]
Changing process priority shouldn't affect the manager, certainly not if you're adjusting the chia.exe processes.

If you close the powershell/command window, it will only terminate manager.py processes started in that environment, the plots will continue normally.  However, when you restart the manager, it loses track of temp directories among other info, and has the ability to overlap new jobs onto tmp locations that already have a job running.

For that reason, I try to avoid restarting the manager.py process under any circumstances.  And instead of using the view and allowing it to keep running, I execute the command and immediately CTRL-C it to terminate right after it displays new stats.  That keeps it from crashing and mucking things up.

**Comment by bathrobehero** [comment]
> However, when you restart the manager, it loses track of temp directories among other info, and has the ability to overlap new jobs onto tmp locations that already have a job running.

As many issues I have, that never happened to me and I frequently restart the service. Maybe you have some problem with the logs (maybe network drive) or something.

But seeing the issues around it seems the project is not updated anymore.

**Comment by Jaga-Telesin** [comment]
Nope, no issue with logs or the network.  It happens on multiple machines (both W10 and W7).  And other people have reported it now, confirming it.  Additionally, the docs for the manager indicate restarting it will clear Tmp2 and other info - I just happened to discover it also clears Tmp1 status/location on tracked jobs.

We'll see if Swar makes a re-appearance.  If not, there are other ways to manage jobs.

**Comment by vc1cv1** [comment]
Are you running the manager in powershell or CMD? if the former, try running it under CMD.

**Comment by Jaga-Telesin** [comment]
Powershell, on both rigs.  Shouldn't have any weaknesses compared to a command prompt.

Due to not wanting to encounter it again on either rig, I will probably steer clear of trying to replicate it, and just keep plotting for now.  Every time it happens I lose around 2/3 of a day's worth of plotting.  Going to pass on that particular pleasure.  :)
